src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+
+        private static ImmutableArray<MemberDeclarationSyntax> GetModifiedMembersForPositionalRecord(
+            INamedTypeSymbol type,
+            TypeDeclarationSyntax classDeclaration,
+            SemanticModel semanticModel,
+            ImmutableArray<PropertyDeclarationSyntax> propertiesToMove,
+            CancellationToken cancellationToken)
+        {
+            var oldMembers = classDeclaration.Members;
+            // create capture variables for the equality operators, since we want to check both of them at the same time
+            // to make sure they can be deleted
+            OperatorDeclarationSyntax? equals = null;
+            OperatorDeclarationSyntax? notEquals = null;
+
+            using var _ = ArrayBuilder<MemberDeclarationSyntax>.GetInstance(out var modifiedMembers);
+            foreach (var member in oldMembers)
+            {
+                switch (member)
+                {
+                    case PropertyDeclarationSyntax property:
+                        {
+                            if (propertiesToMove.Contains(property))
+                            {
+                                continue;
+                            }
+                            break;
+                        }
+
+                    case ConstructorDeclarationSyntax constructor:
+                        {
+                            // remove the constructor with the same parameter types in the same order as the positional parameters
+                            var constructorParamTypes = constructor.ParameterList.Parameters.SelectAsArray(p => p.Type!.ToString());
+                            var positionalParamTypes = propertiesToMove.SelectAsArray(p => p.Type!.ToString());
+                            if (constructorParamTypes.SequenceEqual(positionalParamTypes))
+                            {
+                                if (!IsSimpleConstructor(
+                                    constructor, propertiesToMove, constructor.ParameterList.Parameters, semanticModel, cancellationToken))
+                                {
+                                    // if it is too complex to remove (param validation, error handling, side effects, not everything assigned)
+                                    // we won't delete, and give a warning
+                                    // TODO: if parameter defaults in constructor, move those to primary constructor
+                                    // TODO: Change warning string
+                                    modifiedMembers.Add(
+                                        constructor.WithAdditionalAnnotations(WarningAnnotation.Create("Placeholder Constructor warning")));
+                                }
+                                continue;
+                            }
+
+                            if (constructorParamTypes.Length == 1 &&
+                                constructorParamTypes.First() == classDeclaration.Identifier.ToString())
```

Comment:
what is this check even trying to do?  seems... suspect.

Reply:
I was trying to check for a copy constructor by seeing if there was a constructor with a single param with type equal to the current class type. Since I don't actually do anything with it I'll remove it for now, and when I get around to handling it I'll compare the type symbols directly instead of the names.